drivers/storage/portworx/component/security.go

```diff
@@ -0,0 +1,265 @@
+	SecurityPXSystemSecretName = "px-system"
+	// SecurityPXAuthSecretKey is the data key for any secret containing an auth secret
+	SecurityPXAuthSecretKey = "auth-secret"
+	// SecurityEnvKeyPortworxAuthSystemKey is the environment variable name for the PX security secret
+	SecurityEnvKeyPortworxAuthSystemKey = "PORTWORX_AUTH_SYSTEM_KEY"
+	// SecurityEnvPortworxAuthJwtSharedSecret is an environment variable defining the PX Security JWT secret
+	SecurityEnvPortworxAuthJwtSharedSecret = "PORTWORX_AUTH_JWT_SHAREDSECRET"
+	// SecurityEnvPortworxAuthJwtIssuer is an environment variable defining the PX Security JWT Issuer
+	SecurityEnvPortworxAuthJwtIssuer = "PORTWORX_AUTH_JWT_ISSUER"
+)
+
+type security struct {
+	k8sClient client.Client
+}
+
+// Initialize initializes the componenet
+func (c *security) Initialize(
+	k8sClient client.Client,
+	k8sVersion version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	c.k8sClient = k8sClient
+}
+
+// IsEnabled checks if the components needs to be enabled based on the StorageCluster
+func (c *security) IsEnabled(cluster *corev1alpha1.StorageCluster) bool {
+	return cluster.Spec.Security != nil && *cluster.Spec.Security.Enabled
+}
+
+// Reconcile reconciles the component to match the current state of the StorageCluster
+func (c *security) Reconcile(cluster *corev1alpha1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	err := c.setJwtIssuer(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	err = c.createSystemSecret(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	err = c.createAdminSecret(cluster, ownerRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Delete deletes the component if present
+func (c *security) Delete(cluster *corev1alpha1.StorageCluster) error {
+	err := c.deleteSystemSecret(cluster)
+	if err != nil {
+		return err
+	}
+
+	err = c.deleteAdminSecret(cluster)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MarkDeleted marks the component as deleted in situations like StorageCluster deletion
+func (c *security) MarkDeleted() {
+
+}
+
+func (c *security) setJwtIssuer(
+	cluster *corev1alpha1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	for i := range cluster.Spec.Env {
+		if cluster.Spec.Env[i].Name == SecurityEnvPortworxAuthJwtIssuer {
+			value, err := pxutil.GetValueFromEnv(context.TODO(), c.k8sClient, &cluster.Spec.Env[i], cluster.Namespace)
+			if err != nil {
+				return err
+			}
+
+			// If value does not equal spec value, update to match the spec
+			if value != *cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer {
+				cluster.Spec.Env[i].Value = *cluster.Spec.Security.Auth.Authenticators.SelfSigned.Issuer
+				cluster.Spec.Env[i].ValueFrom = nil
+			}
+
+			// found and updated, exit
+			return nil
+		}
+	}
+
+	// if not found, add default jwt issuer
+	cluster.Spec.Env = append(cluster.Spec.Env, v1.EnvVar{
```

Comment:
Instead of setting back the env, I think we should just create and pass the env directly to the pod in deployment.go
We should use the value in selfSigned.Issuer as it will always be present.

Basically we won't need this setJwtIssuer method.

Reply:
Fixed this